test/analysis_tests.py

```diff
@@ -158,6 +158,32 @@ def test_unormalize_outputs(self):
 
 		# Get rid of the file we made
 		os.remove(input_norm_path)
+		
+	def test_normalize_mu_prec(self):
+		# Need to test that the precision matrices output are symmetric!
+		metadata = pd.read_csv(self.fake_test_folder + 'metadata.csv')
+		learning_params = ['main_deflector_parameters_theta_E',
+			'main_deflector_parameters_gamma1','main_deflector_parameters_gamma2',
+			'main_deflector_parameters_gamma']
+		input_norm_path = self.fake_test_folder + 'norms.csv'
+		norm_dict = Analysis.dataset_generation.normalize_outputs(metadata,
+			learning_params,input_norm_path)
+		
+        # Actual mean/precision matrix output by a network for these params
+		proposal_means = np.asarray([ 0.80717194, -0.00500261,  0.13598095,  2.0799267 ]).astype(np.float64)
+		proposal_prec = np.asarray([[135114.03  ,  35584.293 ,  -6538.843 ,  -1039.7645],
+            [ 35584.293 ,  43250.586 ,  -1896.0295, -11420.458 ],
+            [ -6538.843 ,  -1896.0295,  62808.117 ,    947.9287],
+            [ -1039.7645, -11420.458 ,    947.9287,   6692.047 ]]).astype(np.float64)
+		
+		mu,prec = Analysis.dataset_generation.normalize_mu_prec(proposal_means,
+			proposal_prec,input_norm_path)
+		
+		# Testing for symmetry of precision matrix
+		self.assertTrue(np.allclose(prec, prec.T))
```

Comment:
Is the only check that the precision matrix is symmetric? Could be worth confirming that the normalization was applied correctly.